data-prepper-api/src/main/java/org/opensearch/dataprepper/expression/ExpressionEvaluator.java

```diff
@@ -20,9 +20,16 @@
      *
      * @param statement string to be parsed and evaluated
      * @param context event used to resolve external references in the statement
-     * @return coerced result of statement evaluation
-     *
-     * @throws ExpressionEvaluationException if unable to evaluate or coerce the statement result to type T
+     * @return result of statement evaluation
      */
-    T evaluate(final String statement, final Event context);
+    Object evaluate(final String statement, final Event context);
+
+    default Boolean evaluateConditional(final String statement, final Event context) {
+        final Object result = evaluate(statement, context);
+        if (result instanceof Boolean) {
+            return (Boolean) result;
+        } else {
+            throw new ClassCastException("Unexpected expression return type of " + result.getClass());
```

Comment:
I took a look at the `ParseTreeCoercionService` class. It appears that it will throw the following when it fails to get the correct value.

```
throw new ExpressionCoercionException("Unable to cast " + obj.getClass().getName() + " into " + clazz.getName());
```

Perhaps, we should move this `evaluateConditional` implementation into the `GenericExpressionEvaluator` and use that `ParseTreeCoercionService` class instead.

So this would be an unimplemented method here in the interface.

```
evaluateConditional(String statement, Event context);
```

Then the `GenericExpressionEvaluator` looks somewhat like:


```
@Override
public Boolean evaluateConditional(final String statement, final Event context) {
    final Object result = evaluate(statement, context);
    return coersionService.coerce(result, Boolean.class);
}
```

Reply:
Yes, I could change `ClassCastException` to `ExpressionCoercionException` if that helps. Ultimately there is not much difference between the two when the type is `Boolean` because Boolean cannot be Coerced to a different type anyways. I don't see the need to move the `evaluateConditional` to `GenericExpressionEvaluator` just for that purpose because Boolean is not really coercion friendly anyways. Let me know if you strongly prefer calling coersionService here.